packages/errors/lib/operational-error.js

```diff
@@ -46,8 +55,23 @@ class OperationalError extends Error {
 		if (typeof data.code === 'string') {
 			this.code = OperationalError.normalizeErrorCode(data.code);
 		}
+
+		for (const [key, value] of Object.entries(data)) {
```

Comment:
nitpick: I _feel_ too procedural this piece of code.
Is this wrong at some point?
`this.data = Object.assign(data, ...(this.constructor.reservedKeys.map(key=>this[key])));`

Reply:
I did look at using `Object.assign` however if you look at the condition in this `for` loop you'll see it's negated, which makes the alternative a bit more complex. I can't find a simpler way to achieve the same resulting data with `Object.assign` or array iteration methods, happy if you can find something!

I've got this:

```js
this.data = Object.fromEntries(
	Object.entries(data).filter(
		// @ts-ignore TypeScript does not properly infer the constructor
		([key]) => !this.constructor.reservedKeys.includes(key)
	)
);
```

Which (once Prettier enforces code style) is the same number of lines as the `for of` except I think it's far less easy to understand at first glance.